src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.SkeletonReferenceSet.cs

```diff
@@ -14,7 +14,17 @@ internal partial class SolutionState
 {
     private sealed class SkeletonReferenceSet
     {
-        private readonly ITemporaryStreamStorage _storage;
+        /// <summary>
+        /// The actual assembly metadata produced from another compilation.
+        /// </summary>
+        private readonly AssemblyMetadata _metadata;
+
+        /// <summary>
+        /// Used to tie lifetime of the underlying direct memory to any metadata reference we pass out.  This way the
+        /// memory will not be GC'ed while this reference is alive.
+        /// </summary>
+        private readonly ISupportDirectMemoryAccess? _directMemoryAccess;
```

Comment:
the reference set type is now very concrete.  it holds the real computed data in a strong fashion, and allows it to be found associated both with compilation-trackers (which pass the set forward), or associated with compilations, in case we have compilation-reuse across trackers.

Reply:
this type is now exceedingly trivial as well, as all it does it hold onto this raw data and then create/cache using it to create the actual PEReference objects that are handed out.